src/libxively/tls/xi_tls_layer.c

```diff
@@ -275,18 +280,6 @@ static xi_state_t recv_handler( void* context, void* data, xi_state_t in_out_sta
         return XI_STATE_OK;
     }
 
-    /* if we suppose to reuse the buffer */
-    if ( XI_STATE_WANT_READ == in_out_state && NULL != data_desc )
-    {
```

Comment:
here my question is regarding the 1st:
As I see in xi_tls_layer.c the layer_data->raw_buffer is filled when package arrived on socket. How will TLS library know that this layer_data->raw_buffer buffer contains more without interacting with it?
layer_data->raw_buffer is a list. Suppose there are 2 elements in the list. TLS layer reads the first element fully. And this element is either one or more full TLS packages. Which means TLS package does not span over the border of the two list elements. How will TLS library know that the raw_buffer contains another element?